applications/RANSApplication/custom_processes/rans_apply_exact_nodal_periodic_condition_process.cpp

```diff
@@ -0,0 +1,321 @@
+
+            if (static_cast<std::size_t>(r_master_patch_index) != r_master_node.Id()) {
+                std::vector<std::size_t> node_id_list = {
+                    r_master_node.Id(), static_cast<std::size_t>(r_master_patch_index)};
+#pragma omp critical
+                {
+                    condition_id++;
+                    auto p_condition = r_base_model_part.CreateNewCondition(
+                        "PeriodicCondition", condition_id, node_id_list, p_properties);
+                    p_condition->Set(PERIODIC);
+                    r_master_node.Set(PERIODIC);
+                }
+            }
+        }
+    } else if (mRotationAngle > eps) {
+#pragma omp parallel for shared(condition_id)
+        for (int i_node = 0; i_node < number_of_nodes; ++i_node) {
+            auto& r_master_node = *(r_master_model_part_nodes.begin() + i_node);
+            const array_1d<double, 3>& master_initial_position =
+                r_master_node.GetInitialPosition().Coordinates();
+            const array_1d<double, 3>& master_final_position =
+                CalculateRotatedPosition(master_initial_position);
+            int& r_master_patch_index = r_master_node.FastGetSolutionStepValue(PATCH_INDEX);
+            for (int j_node = 0; j_node < number_of_nodes; ++j_node) {
+                const auto& r_slave_node = *(r_slave_model_part_nodes.begin() + j_node);
+                if (norm_2(master_final_position -
+                           r_slave_node.GetInitialPosition().Coordinates()) < mTolerance) {
+                    r_master_patch_index = r_slave_node.Id();
+                    break;
+                }
+            }
+
+            KRATOS_ERROR_IF(r_master_patch_index == 0)
+                << "Slave node is not found in " << mSlaveModelPartName
+                << " near " << master_final_position << " with tolerance "
+                << mTolerance << " for master node id=" << r_master_node.Id()
+                << " at " << r_master_node.GetInitialPosition().Coordinates()
+                << " in " << mMasterModelPartName << ".\n";
+
+            if (static_cast<std::size_t>(r_master_patch_index) != r_master_node.Id()) {
+                std::vector<std::size_t> node_id_list = {
+                    r_master_node.Id(), static_cast<std::size_t>(r_master_patch_index)};
+#pragma omp critical
+                {
+                    condition_id++;
+                    auto p_condition = r_base_model_part.CreateNewCondition(
+                        "PeriodicCondition", condition_id, node_id_list, p_properties);
+                    p_condition->Set(PERIODIC);
+                    r_master_node.Set(PERIODIC);
+                }
+            }
+        }
+    } else if (mTranslationMagnitude > eps) {
+#pragma omp parallel for shared(condition_id)
+        for (int i_node = 0; i_node < number_of_nodes; ++i_node) {
+            auto& r_master_node = *(r_master_model_part_nodes.begin() + i_node);
```

Comment:
I think the dereference is not required here.

Reply:
moved to lambda functions :), so less code duplication.